### ISSUE TITLE
Integrating all the dygraphs options

### ISSUE THREAD

**Issue by janga1997** [dinkelk/PyDyGraphs#7]
Since there are too many options in **dygraphs** to just cram into the **plot** function, the way the R library deals with this is, 
1. Make a minimal set of options available in the main plot function, namely
* data
* xlabels
* ylabels
* width
* height

For the rest of the customizations, they can just call the `dyOptions` class.

```
dygraph(ldeaths, main = "Deaths from Lung Disease (UK)") %>%
  dyOptions(drawPoints = TRUE, pointSize = 5, pointShape = "triangle")
```
I was thinking of doing something similar. 

```
import dygraphs.graph as dy
from dygraphs import Options

# Single Figure Example
fig = dy.figure()

# Plot figure
fig.plot(x,y)
fig.title("Figure 1")
fig.xlabel('Series')
fig.ylabel('Value')

**Set Options**
fig.options(Options(stepPlot=True, logScale=True, fillGraph=True))

# Show figure:
fig.show()
```

**Comment by dinkelk** [comment]
In general, I am not opposed to that idea. However, I have never seen a python plotting API that uses the pattern of passing complex objects as arguments regularly. Matplotlib, for instance has a plot function with the definition:

```
matplotlib.pyplot.plot(*args, **kwargs)
```

See full docs [here](https://matplotlib.org/api/pyplot_api.html#matplotlib.pyplot.plot). The user can pass a myriad of options (>20) to the plot function itself.

From a user perspective, I think that either is pretty easy to use, although I am more used to seeing the matplotlib `kwargs` approach. From an implementation perspective I certainly like the idea of having an `Options` object that encapsulates all of that complexity.

Are you opposed to this idea? We provide the user a plot function similar to matplotlib using `(*args, **kwargs)` with a good set of documentation describing each optional argument. And then internally, we deal with all the arguments by passing `(*args, **kwargs)` to an `Options` class which handles the complexity of organizing and generating the javascript necessary for the dygraphs options. The user would never see (or need to instantiate) that object.

I actually still like the idea of the `fig.options()` call there. I just think it is a little odd (or maybe just unpythonic in relation to what I have seen) to see the `Options()` object instantiated and passed in. 

Why not just give the user the option to either:

```
fig.plot(x, y)
fig.options(stepPlot=True, logScale=True, fillGraph=True)
```
or
```
fig.plot(x, y, stepPlot=True, logScale=True, fillGraph=True)
```

**Comment by janga1997** [comment]
Actually, I like the last option the best.
`fig.plot(x, y, stepPlot=True, logScale=True, fillGraph=True)`

Internally, it would be represented as 
```
def plot(x, y, **kwargs):
................
```
But we would also have to keep the variables names same as they are in the dygraphs.js

That means, `logscale` and `rangeselector` should be `logScale` and `rangeSelector`,
and `labels` instead of `ylabels`, where the first element in labels is the label for the x-variable.

Then we can capture all the arguments passed to `**kwargs` in a single statement.

```
import json
self._options = json.dumps(kwargs)
```
and pass `options` to the javascript string.

**Comment by dinkelk** [comment]
Perfect. That solution is both clean to implement and elegant to use.

**Comment by janga1997** [comment]
@dinkelk Also, I think we can ditch pandas as a dependancy.
We still would have `numpy` as a dependency, but that should be okay. Even matplotlib uses it.
I will push a commit with all these changes soon.

**Comment by dinkelk** [comment]
@janga1997 Awesome! I have been wanting to remove that for a long time.

**Comment by dinkelk** [comment]
Resolved in db5e281b4b77a8907732cb62ff25b3045cde457c.
